a10_octavia/tests/unit/controller/worker/tasks/test_a10_network_tasks.py

```diff
@@ -0,0 +1,382 @@
+from a10_octavia.controller.worker.tasks import a10_network_tasks
+from a10_octavia.tests.common import a10constants
+from a10_octavia.tests.unit import base
+
+MEMBER = o_data_models.Member()
+VTHUNDER = data_models.VThunder()
+SUBNET = o_net_data_models.Subnet()
+PORT = o_net_data_models.Port()
+VRID = data_models.VRID()
+
+
+class MockIP(object):
+    def __init__(self, ip_address):
+        self.ip_address = ip_address
+
+
+class TestNetworkTasks(base.BaseTaskTestCase):
+
+    def setUp(self):
+        super(TestNetworkTasks, self).setUp()
+        imp.reload(a10_network_tasks)
+        patcher = mock.patch(
+            'a10_octavia.controller.worker.tasks.a10_network_tasks.BaseNetworkTask.network_driver')
+        self.network_driver_mock = patcher.start()
+        self.client_mock = mock.Mock()
+
+    # 1 no vrid and no floating ip
+    @mock.patch('a10_octavia.common.utils.get_vrid_floating_ip_for_project', return_value=None)
+    def test_handle_vrrp_fip_with_no_vrid_no_fip(self, mock_utils):
+        network_task = a10_network_tasks.HandleVRIDFloatingIP()
+        result = network_task.execute(VTHUNDER, MEMBER, None)
+        self.assertEqual(result, None)
+
+    # 2.a (SHARED_PARTITION) NO VRIP, FIP provided
+    @mock.patch('a10_octavia.common.utils.get_vrid_floating_ip_for_project',
+                return_value=a10constants.MOCK_VRID_FLOATING_IP_1)
+    @mock.patch('a10_octavia.common.utils.get_patched_ip_address',
+                return_value=a10constants.MOCK_VRID_FLOATING_IP_1)
+    def test_handle_vrid_fip_with_no_vrid_but_fip_given_in_shared_partition(
+            self, mock_patched_ip, mock_floating_ip):
+        member = copy.deepcopy(MEMBER)
+        member.subnet_id = a10constants.MOCK_SUBNET_ID
+        subnet = copy.deepcopy(SUBNET)
+        subnet.cidr = a10constants.MOCK_SUBNET_CIDR
+        port = copy.deepcopy(PORT)
+        port.fixed_ips.append(MockIP(a10constants.MOCK_VRID_FLOATING_IP_1))
+        mock_network_task = a10_network_tasks.HandleVRIDFloatingIP()
+        mock_network_task.axapi_client = self.client_mock
+        self.network_driver_mock.get_subnet.return_value = subnet
+        self.network_driver_mock.create_port.return_value = port
+        mock_network_task.execute(VTHUNDER, member, None)
+        self.network_driver_mock.create_port.assert_called_with(
+            subnet.network_id, member.subnet_id, fixed_ip=a10constants.MOCK_VRID_FLOATING_IP_1)
+        self.client_mock.vrrpa.update.assert_called_with(
+            0, floating_ip=a10constants.MOCK_VRID_FLOATING_IP_1)
+
+    # 2.b (SPECIFIC PARTITION) NO VRID, FIP provided
+    @mock.patch('a10_octavia.common.utils.get_vrid_floating_ip_for_project',
+                return_value=a10constants.MOCK_VRID_FLOATING_IP_1)
+    @mock.patch('a10_octavia.common.utils.get_patched_ip_address',
+                return_value=a10constants.MOCK_VRID_FLOATING_IP_1)
+    def test_handle_vrid_fip_with_no_vrid_but_fip_given_in_specific_partition(
```

Comment:
If we have to decide between test names being about inputs or expected outcomes, we should tend towards the expected outcomes

Ideally `test_ClassName/function_name_operation_input` or `test_ClassName/function_name_input_operation`

`test_HandleVRIDFloatingIP_create_floating_ip_in_specified_partition_with_static_ip`

`test_HandleVRIDFloatingIP_with_static_ip_and_specified_parition_create_floating_ip_in_specified_partition`

There seems to be some divide on test names. But the main consensus is that they should tell you exactly what your testing, what goes in, what _should_ come out.

I think for most of our test cases `test_ClassName/function_name_operation_input` will look a little cleaner and be a bit shorter